src/lib.rs

```diff
@@ -191,6 +204,22 @@ pub fn set_handler<F>(user_handler: F) -> Result<(), Error>
     }
 
     thread::spawn(move || {
+        // block the signals so it reliably delivers to other threads
+        #[cfg(all(feature="norestart",unix))]
+        {
+            extern crate libc;
+            unsafe {
+                let mut sigset = ::std::mem::uninitialized::
+                    <libc::sigset_t>();
+                libc::sigemptyset(&mut sigset);
+                libc::sigaddset(&mut sigset, libc::SIGINT);
+                #[cfg(feature = "termination")]
+                libc::sigaddset(&mut sigset, libc::SIGTERM);
+                libc::pthread_sigmask(
+                    libc::SIG_BLOCK, &sigset, ::std::ptr::null_mut());
+            }
```

Comment:
Why do we need to do this? This will block the signals on this thread only, right?